howto/serialize/README.md

```diff
@@ -0,0 +1,138 @@
+# Serialization in Dapr's SDKs
+
+An SDK for Dapr should provide serialization for two use cases. First, for API objects sent through request and response payloads. Second, for objects to be persisted. For both these use cases, a default serialization is provided. In the Java SDK, it is the [DefaultObjectSerializer](https://dapr.github.io/java-sdk/io/dapr/serializer/DefaultObjectSerializer.html) class, providing JSON serialization.
+
+## Service invocation
+
+```java
+    DaprClient client = (new DaprClientBuilder()).build();
+    client.invokeService(Verb.POST, "myappid", "saySomething", "My Message", null).block();
+```
+
+In the example above, the app will receive a `POST` request for the `saySomething` method with the request payload as `"My Message"` - quoted since the serializer will serialize the input String to JSON.
+
+```text
+POST /saySomething HTTP/1.1
+Host: localhost
+Content-Type: text/plain
+Content-Length: 12
+
+"My Message"
+```
+
+## State management
+
+```java
+    DaprClient client = (new DaprClientBuilder()).build();
+    client.saveState("MyStateStore", "MyKey", "My Message").block();
+```
+In this example, `My Message` will be saved. It is not quoted because Dapr's API will internally parse the JSON request object before saving it.
+
+```JSON
+[
+    {
+        "key": "MyKey",
+        "value": "My Message"
+    }
+]
+```
+
+## PubSub
+
+```java
+  DaprClient client = (new DaprClientBuilder()).build();
+  client.publishEvent("TopicName", "My Message").block();
+```
+
+The event is published and the content is serialized to `byte[]` and sent to Dapr sidecar. The subscriber will receive it as a [CloudEvent](https://github.com/cloudevents/spec). Cloud event defines `data` as String. Dapr SDK also provides a built-in deserializer for `CloudEvent` object.
+
+```java
+  @PostMapping(path = "/TopicName")
+  public void handleMessage(@RequestBody(required = false) byte[] body) {
+      // Dapr's event is compliant to CloudEvent.
+      CloudEvent event = CloudEvent.deserialize(body);
+  }
+```
+
+## Bindings
+
+In this case, the object is serialized to `byte[]` as well and the input binding receives the raw `byte[]` as-is and deserializes it to the expected object type.
+
+* Output binding:
+```java
+    DaprClient client = (new DaprClientBuilder()).build();
+    client.invokeBinding("sample", "My Message").block();
+```
+
+* Input binding:
+```java
+  @PostMapping(path = "/sample")
+  public void handleInputBinding(@RequestBody(required = false) byte[] body) {
+      String message = (new DefaultObjectSerializer()).deserialize(body, String.class);
+      System.out.println(message);

```

Comment:
It's helpful to the user if all the output of all "prints" are shown like in https://gobyexample.com/hello-world